device-registry/src/main/scala/org/genivi/sota/device_registry/DevicesResource.scala

```diff
@@ -47,12 +47,14 @@ class DevicesResource(namespaceExtractor: Directive1[AuthedNamespaceScope],
 
   def searchDevice(ns: Namespace): Route =
     parameters(('regex.as[String Refined Regex].?,
-                'deviceId.as[String].?)) { // TODO: Use refined
-      case (Some(re), None) =>
-        complete(db.run(DeviceRepository.search(ns, re)))
-      case (None, Some(deviceId)) =>
+                'deviceId.as[String].?,
+                'offset.as[Long] ? 0,
+                'limit.as[Long] ? 50)) { // TODO: Use refined
+      case (Some(re), None, offset, limit) =>
+        complete(db.run(DeviceRepository.search(ns, re, offset, limit)))
+      case (None, Some(deviceId), _, _) =>
         complete(db.run(DeviceRepository.findByDeviceId(ns, DeviceId(deviceId))))
-      case (None, None) => complete(db.run(DeviceRepository.list(ns)))
+      case (None, None, offset, limit) => complete(db.run(DeviceRepository.list(ns, offset, limit)))
```

Comment:
Yeah that makes sense. Unfortunately, adding that change requires changing DeviceRegistryClient.scala, and this breaks multiple endpoints in resolver, which try and list all devices in device registry. The broken endpoints in resolver will be obsolete once we migrate the data from resolver. Jerry suggested removing the endpoints and adding a `SearchResult` object from device registry in a later ticket. For now, device registry should return pages only if explicitly requested. Make sense?